File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConfiguringLoggers_NLog.vb
Imports LogMan = NLog.LogManager
Imports NLog.LogManager

Namespace Logging
    Class NLogLogging
        ' RSPEC-4792: https://jira.sonarsource.com/browse/RSPEC-4792
        Private Sub Foo(ByVal config As NLog.Config.LoggingConfiguration)
            NLog.LogManager.Configuration = config
'           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^   {{Make sure that this logger's configuration is safe.}}
        End Sub


        Private Sub AdditionalTests(config As NLog.Config.LoggingConfiguration)
            ' Call via the alias
            LogMan.Configuration = config
'           ^^^^^^^^^^^^^^^^^^^^   {{Make sure that this logger's configuration is safe.}}

            ' Call via the import of the static class
            Configuration = config
'           ^^^^^^^^^^^^^   {{Make sure that this logger's configuration is safe.}}


            ' Reading the configuration should be ok
            Dim current = Configuration
            current = LogMan.Configuration
            current = NLog.LogManager.Configuration

            AdditionalTests(Configuration)
            AdditionalTests(LogMan.Configuration)
            AdditionalTests(NLog.LogManager.Configuration)
        End Sub
    End Class
End Namespace

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ConfiguringLoggers_NLog.cs
using LogMan = NLog.LogManager;
using static NLog.LogManager;

namespace MvcApp
{
    class NLogLogging
    {
        // RSPEC-4792: https://jira.sonarsource.com/browse/RSPEC-4792
        void Foo(NLog.Config.LoggingConfiguration config)
        {
            NLog.LogManager.Configuration = config;
//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^   {{Make sure that this logger's configuration is safe.}}
        }


        void AdditionalTests(NLog.Config.LoggingConfiguration config)
        {
            // Call via the alias
            LogMan.Configuration = config;
//          ^^^^^^^^^^^^^^^^^^^^   {{Make sure that this logger's configuration is safe.}}

            // Call via the "using static"
            Configuration = config;
//          ^^^^^^^^^^^^^   {{Make sure that this logger's configuration is safe.}}


            // Reading the configuration should be ok
            var current = Configuration;
            current = LogMan.Configuration;
            current = NLog.LogManager.Configuration;

            AdditionalTests(Configuration);
            AdditionalTests(LogMan.Configuration);
            AdditionalTests(NLog.LogManager.Configuration);

        }
    }
}
